repositories/MerchOrderRepository.ts

```diff
@@ -40,12 +40,28 @@ export class MerchOrderRepository extends BaseRepository<OrderModel> {
   }
 
   /**
-   * Gets all orders for a given user. Returns the order joined with its pickup event.
+   * Gets all orders for a given user. Returns the order joined with its pickup event and user.
    */
   public async getAllOrdersForUser(user: UserModel): Promise<OrderModel[]> {
     return this.repository.find({ user });
   }
 
+  /**
+   * Gets all orders for a given user. Returns the order joined with its pickup event, user,
+   * merch item options, and merch items.
+   */
+  public async getAllOrdersWithItemsForUser(user: UserModel): Promise<OrderModel[]> {
+    return this.repository
+      .createQueryBuilder('order')
+      .leftJoinAndSelect('order.pickupEvent', 'orderPickupEvent')
+      .leftJoinAndSelect('order.items', 'orderItem')
+      .leftJoinAndSelect('order.user', 'user')
+      .leftJoinAndSelect('orderItem.option', 'option')
+      .leftJoinAndSelect('option.item', 'merchItem')
+      .where('order.user = :uuid', { uuid: user.uuid })
+      .getMany();
+  }
```

Comment:
This new query is needed so that the join happening in the following code is possible: 
https://github.com/acmucsd/membership-portal/pull/271/files#diff-8f5e873f32fd041a327e36c028853a65ced08b99cb09572d3d785ef193b0c212R804-R806